File: daily/2025/2962_Count_Subarrays_Where_Max_Element_Appears_at_Least_K_Times.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {
        int M = *max_element(nums.begin(), nums.end());
        vector<int> index;
        long long ans = 0;
        for (int i = 0; i < nums.size(); i++){
            if (nums[i] == M) index.push_back(i);
            int freq = index.size();
            if (freq >= k){
                ans += index[freq - k] + 1;
            }
        }
        return ans;
    }
};

int main(){
    vector<int> nums = {1,3,2,3,3};
    int k = 2;
    // vector<int> nums = {1,4,2,1};
    // int k = 3;
    Solution S;
    cout << S.countSubarrays(nums, k) << endl;
    return 0;
}

File: daily_python/2962_Count_Subarrays_Where_Max_Element_Appears_at_Least_K_Times.py
import os
import sys
from typing import List

class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        M = max(nums)
        curr = 0
        ans = 0
        start = 0
        for end in range(n):
            if nums[end] == M:
                curr += 1
            while curr == k:
                if nums[start] == M:
                    curr -= 1
                start += 1
            ans += (start - 1) + 1
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            k = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((nums, k))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums, k in dataset:
        results.append(solution.countSubarrays(nums, k))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')
